packages/pcset/index.d.ts

```diff
@@ -0,0 +1,127 @@
+type Binary = "0" | "1";
+type Chroma = Binary[12];
+type Interval = string;
+/**
+ * Get chroma of a pitch class set. A chroma identifies each set uniquely.
+ * It"s a 12-digit binary each presenting one semitone of the octave.
+ *
+ * Note that this function accepts a chroma as parameter and return it
+ * without modification.
+ *
+ * @param {Array|String} set - the pitch class set
+ * @return {String} a binary representation of the pitch class set
+ * @example
+ * PcSet.chroma(["C", "D", "E"]) // => "1010100000000"
+ */
+export declare function chroma(set: string | string[]): Chroma;
+/**
+ * Get a list of all possible chromas (all possible scales)
+ * More information: http://allthescales.org/
+ * @return {Array} an array of possible chromas from '10000000000' to '11111111111'
+ *
+ */
+export declare function chromas(numberOfNotes: number): Chroma[];
+/**
+ * Given a a list of notes or a pcset chroma, produce the rotations
+ * of the chroma discarding the ones that starts with "0"
+ *
+ * This is used, for example, to get all the modes of a scale.
+ *
+ * @param {Array|String} set - the list of notes or pitchChr of the set
+ * @param {Boolean} normalize - (Optional, true by default) remove all
+ * the rotations that starts with "0"
+ * @return {Array<String>} an array with all the modes of the chroma
+ *
+ * @example
+ * PcSet.modes(["C", "D", "E"]).map(PcSet.intervals)
+ */
+export declare function modes(
+  set: string | string[],
+  normalize?: boolean
+): string[];
+/**
+ * Test if the given string is a pitch class set chroma.
+ * @param {String} chroma - the pitch class set chroma
+ * @return {Boolean} true if its a valid pcset chroma
+ * @example
+ * PcSet.isChroma("101010101010") // => true
+ * PcSet.isChroma("101001") // => false
+ */
+export declare function isChroma(set: string): boolean;
```

Comment:
I know there's a way in typescript that some function _ensures_ that a thing is a type. This one ensures the string is a ChromaString